authors/apps/authentication/tests/test_login.py

```diff
@@ -0,0 +1,48 @@
+""" module to test login. """
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from authors.apps.authentication.models import User
+
+
+class AuthenticationTests(APITestCase):
+    def setUp(self):
+        """ Setup data for the tests """
+        self.user1 = {
+            "username": "createduser1",
+            "email": "createduser1@user.user",
+            "password": "user123user"
+        }
+        self.reg_url = reverse('authentication:register')
+        self.login_url = reverse('authentication:login')
+        reg = self.client.post(self.reg_url, self.user1, format='json')
+
+    def test_successful_login_user(self):
+        """ Test that a user successfully logs in """
+        response = self.client.post(self.login_url, self.user1, format='json')
+        self.assertEqual(login_response.status_code, status.HTTP_201_CREATED)
+
+    def test_unsuccessful_login_with_wrong_email(self):
+        """ Test that a user cannot log in with a wrong email """
+        self.user1['email'] = "wrong@user.user"
```

Comment:
For a wrong email, I have noticed that you have used **wrong@user.user** which to me seems to be a valid email address format, what do you think of emails in the format **wrongemail**,**wrongemail@wrong**,**@wrongemail@wrong** etc so that all edge cases are taken into account.

Reply:
I understand the concern. The test is to test when a user inputs a valid email but it is not their specific email.